schemas/webpack.d.ts

```diff
@@ -0,0 +1,364 @@
+        hotUpdateFunction?: string;
+
+        // The filename of the Hot Update Main File. It is inside the `output.path` directory.
+        hotUpdateMainFilename?: string;
+
+        // The JSONP function used by webpack for async loading of chunks.
+        jsonpFunction?: string;
+
+        // If set, export the bundle as library. `output.library` is the name.
+        library?: string | string[];
+
+        libraryTarget?: OutputLibraryTarget;
+
+        // The output directory as **absolute path** (required).
+        path?: string;
+
+        // Include comments with information about the modules.
+        pathinfo?: boolean;
+
+        // The `publicPath` specifies the public URL address of the output files when referenced in a browser.
+        publicPath?: string;
+
+        // The filename of the SourceMaps for the JavaScript files. They are inside the `output.path` directory.
+        sourceMapFilename?: string;
+
+        // Prefixes every line of the source in the bundle with this string.
+        sourcePrefix?: string;
+
+        // If `output.libraryTarget` is set to umd and `output.library` is set, setting this to true will name the AMD module.
+        umdNamedDefine?: boolean;
+    }
+
+    interface OutputAuxiliaryComment {
+        amd?: string;
+        commonjs?: string;
+        commonjs2?: string;
+        root?: string;
+    }
+
+    interface OutputDevtoolLineToLine {
+        exclude?: string;
+        include?: string;
+        test?: string;
+    }
+
+    interface Resolve {
+        alias?: { key: string } | ResolveAlias[];
+        aliasFields?: [string | string[]];
+        cachePredicate?: Function;
+        descriptionFiles?: string[];
+        enforceExtension?: boolean;
+        enforceModuleExtension?: boolean;
+        extensions?: string[];
+        fileSystem?: any; // TODO missing type
+        mainFields?: [string | string[]];
+        mainFiles?: string[];
+        moduleExtensions?: string[];
+        modules?: string[];
+        plugins?: any[];
+        resolver?: any; // TODO missing type
+        symlinks?: boolean;
+        unsafeCache?: boolean | Object;
+    }
+
+    interface ResolveAlias {
+        alias?: string;
+        name?: string;
+        onlyModule?: boolean;
+    }
+
+    interface Object {
+        [key: string]: any;
+    }
+
+    type EntryItem = string | string[];
+
+    type OutputLibraryTarget =
+        "var"
+        | "assign"
+        | "this"
+        | "window"
+        | "global"
+        | "commonjs"
+        | "commonjs2"
+        | "commonjs-module"
+        | "amd"
+        | "umd"
+        | "umd2"
+        | "jsonp";
+
+    type Stats =
+        "none"
+        | "errors-only"
+        | "minimal"
+        | "normal"
+        | "verbose";
+
+    type Target =
+        "web"
+        | "webworker"
+        | "node"
+        | "async-node"
+        | "node-webkit"
+        | "electron";
```

Comment:
`"atom"` / `"electron-main"` are (possibly?) aliases to `"electron"`. There is also a separate `"electron-renderer"` target.